### ISSUE TITLE
Ensure markup documents contain well-formed elements

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/exhibits#2497]
### Current behavior

Some elements are not well-formed.

### Steps to reproduce

Example page: [Martin Wong Catalogue Raisonné](https://exhibits.stanford.edu/martin-wong), or any other exhibit.

Input example page(s) to W3C's HTML validator: https://validator.w3.org/nu/.

### Possible solution

WCAG Level A: [H75: Ensuring that Web pages are well-formed | Techniques for WCAG 2.0](https://www.w3.org/TR/WCAG20-TECHS/H75.html).

### Reported by SODA

> Violation: Ensure markup documents contain well-formed elements
> Severity: Minor
> 
> [Issue]
> Some elements are not well formed.
> 
> - Element div not allowed as child of element span
> - Element p not allowed as child of element h2 
> 
> [User Impact]
> When elements are not well formed, assistive technologies may not behave as intended.
> 
> [Recommendation]
> Use the W3C's HTML validator and fix any issues that are presented: 
> https://validator.w3.org/nu/
> 
> Specifically, look at:
> - Duplicate or Invalid IDs
> - Element XX not allowed as child of XX
> - Starting or Eneding Tags
> - Anything related to ARIA or ROLE. 
> - Suppressing further errors

**Comment by jcoyne** [comment]
The only "not allowed as child" is

```
Error: Element [div](https://html.spec.whatwg.org/multipage/#the-div-element) not allowed as child of element [span](https://html.spec.whatwg.org/multipage/#the-span-element) in this context. (Suppressing further errors from this subtree.)

From line 99, column 3; to line 99, column 30

dden;">↩  <div class="form-group row"><label

Contexts in which element [div](https://html.spec.whatwg.org/multipage/#the-div-element) may be used:
    Where [flow content](https://html.spec.whatwg.org/multipage/#flow-content-2) is expected.
    As a child of a [dl](https://html.spec.whatwg.org/multipage/#the-dl-element) element.
Content model for element [span](https://html.spec.whatwg.org/multipage/#the-span-element):
    [Phrasing content](https://html.spec.whatwg.org/multipage/#phrasing-content-2).
```

This corresponds to this HTML:
```html
      <span style="display:none;visibility:hidden;">
  <div class="form-group row"><label class="col-form-label col-sm-3" for="contact_form_email_address">Ignore this text box. It is used to detect spammers. If you enter anything into this text box, your message will not be sent.</label><div class="col-sm-9"><input class="form-control" type="email" name="contact_form[email_address]" id="contact_form_email_address" /></div></div>
</span>
```

**Comment by dnoneill** [comment]
I am working on the meta tags being malformed but the rest of this is up for grabs

**Comment by taylor-steve** [comment]
Let's rerun this after shipping the BL8 branch

**Comment by taylor-steve** [comment]
Looking at some of these now that we have Exhibits on BL8. Apologies in advance any notification spam, looking at these one by one.

![Image](https://github.com/user-attachments/assets/31acba19-4b83-4cba-ba88-e8e75cea6eea)

The spec (and Rails) disagrees: https://html.spec.whatwg.org/multipage/input.html#hidden-state-(type=hidden)

**Comment by taylor-steve** [comment]
![Screenshot 2024-11-26 at 8 14 39 AM](https://github.com/user-attachments/assets/d4336a60-787e-475f-9eed-191f28f7591c)
![Screenshot 2024-11-26 at 8 14 30 AM](https://github.com/user-attachments/assets/f728f45d-0f1f-4441-b06c-d05321f78c49)

I think these are false positives? They are saying you can only use search in the context of flow content and then state that they are indeed being used in the context of flow content...

**Comment by taylor-steve** [comment]
I have https://github.com/ambethia/recaptcha/pull/458 in for the recaptcha markup issue. The rest are duplicates. We could re-run this after the Bootstrap 5 work is in place, with a greater variety of pages? I am slightly skeptical of the output of this tool though.

**Comment by taylor-steve** [comment]
@dnoneill your tophat gem removal has made it to Exhibits, correct? If so, would you mind looking at the meta element results from this scan? I think they are all similar to this:
![Image](https://github.com/user-attachments/assets/65588b31-6aae-4418-a073-fe7c55860c84)
![Image](https://github.com/user-attachments/assets/a4f966c1-1995-499c-bc92-8148787fd514)

**Comment by taylor-steve** [comment]
@dnoneill disregard. It appears it has not been backported to Spotlight 4, so Exhibits doesn't have it yet. Do you recall if that would be possible?

**Comment by dnoneill** [comment]
No. It is a breaking change because we are removing the gem and the gem has a function called in the base.html.erb file.
